src/isl/mod.rs

```diff
@@ -366,75 +373,75 @@ mod isl_tests {
+                2e1.into(),
+                5e1.into()
+            ).unwrap().into()
         ),
         case::range_with_decimal(
             load_range(
                 r#"
                     range::[20.4, 50.5]
                 "#
-            ),
-            Range::decimal_range(
-                RangeBoundaryValue::Value(Decimal::new(204, -1), RangeBoundaryType::Inclusive),
-                RangeBoundaryValue::Value(Decimal::new(505, -1), RangeBoundaryType::Inclusive)
-            )
+            ).unwrap(),
+            DecimalRange::new(
+                Decimal::new(204, -1).into(),
+                Decimal::new(505, -1).into()
+            ).unwrap().into()
         ),
         case::range_with_timestamp(
             load_range(
                 r#"
                     range::[2020-01-01T, 2021-01-01T]
                 "#
-            ),
-            Range::timestamp_range(
-                RangeBoundaryValue::Value(Timestamp::with_year(2020).with_month(1).with_day(1).build().unwrap(), RangeBoundaryType::Inclusive),
-                RangeBoundaryValue::Value(Timestamp::with_year(2021).with_month(1).with_day(1).build().unwrap(), RangeBoundaryType::Inclusive)
-            )
+            ).unwrap(),
+            TimestampRange::new(
+                Timestamp::with_year(2020).with_month(1).with_day(1).build().unwrap().into(),
+                Timestamp::with_year(2021).with_month(1).with_day(1).build().unwrap().into()
+            ).unwrap().into()
         ),
         case::range_with_timestamp_precision(
             load_timestamp_precision_range(
                 r#"
                     range::[year, month]
                 "#
-            ),
-            Range::timestamp_precision_range(
-                RangeBoundaryValue::Value(TimestampPrecision::Year, RangeBoundaryType::Inclusive),
-                RangeBoundaryValue::Value(TimestampPrecision::Month, RangeBoundaryType::Inclusive)
-            )
+            ).unwrap(),
+            TimestampPrecisionRange::new(
+                TimestampPrecision::Year.into(),
+                TimestampPrecision::Month.into()
+            ).unwrap().into()
         ),
         case::range_with_number(
             load_number_range(
                 r#"
                     range::[1, 5.5]
                 "#
-            ),
-            Range::number_range(
-                RangeBoundaryValue::Value((&IntegerValue::I64(1)).into(), RangeBoundaryType::Inclusive),
-                RangeBoundaryValue::Value((&Decimal::new(55, -1)).try_into().unwrap(), RangeBoundaryType::Inclusive)
-            )
+            ).unwrap(),
+            NumberRange::new(
+                Number::from(&IntegerValue::I64(1)).into(),
+                Number::try_from(&Decimal::new(55, -1)).unwrap().into()
+            ).unwrap().into()
         )
     )]
-    fn owned_struct_to_range(range1: IonSchemaResult<Range>, range2: IonSchemaResult<Range>) {
+    fn owned_struct_to_range(range1: Range, range2: Range) {
```

Comment:
I think you could change `range2` to be `impl Into<Range>`, letting you drop the `.into()` calls in each of the `case::`s above.